src/signal.rs

```diff
@@ -206,66 +202,64 @@ mod spin_button {
-        where F: Fn(&SpinButton) -> Inhibit + 'static {
+        where F: Fn(&Self) -> Inhibit + 'static {
             unsafe {
-                let f: Box<Box<Fn(&SpinButton) -> Inhibit + 'static>> =
-                    Box::new(Box::new(output_func));
+                let f: Box<F> = Box::new(output_func);
                 connect_raw(self.to_glib_none().0 as *mut _, b"output\0".as_ptr() as *mut _,
-                        transmute(output_trampoline as usize), Box::into_raw(f) as *mut _)
+                        Some(transmute(output_trampoline::<Self, F> as usize)), Box::into_raw(f))
             }
         }
 
         fn connect_value_changed<F>(&self, value_changed_func: F) -> SignalHandlerId
-        where F: Fn(&SpinButton) + 'static {
+        where F: Fn(&Self) + 'static {
             unsafe {
-                let f: Box<Box<Fn(&SpinButton) + 'static>> =
-                    Box::new(Box::new(value_changed_func));
+                let f: Box<F> = Box::new(value_changed_func);
                 connect_raw(self.to_glib_none().0 as *mut _, b"value-changed\0".as_ptr() as *mut _,
-                        transmute(trampoline as usize), Box::into_raw(f) as *mut _)
+                        Some(transmute(trampoline::<Self, F> as usize)), Box::into_raw(f))
             }
         }
 
         fn connect_wrapped<F>(&self, wrapped_func: F) -> SignalHandlerId
-        where F: Fn(&SpinButton) + 'static {
+        where F: Fn(&Self) + 'static {
             unsafe {
-                let f: Box<Box<Fn(&SpinButton) + 'static>> =
-                    Box::new(Box::new(wrapped_func));
+                let f: Box<F> = Box::new(wrapped_func);
                 connect_raw(self.to_glib_none().0 as *mut _, b"wrapped\0".as_ptr() as *mut _,
-                        transmute(trampoline as usize), Box::into_raw(f) as *mut _)
+                        Some(transmute(trampoline::<Self, F> as usize)), Box::into_raw(f))
             }
         }
     }
 
-    unsafe extern "C" fn change_trampoline(this: *mut GtkSpinButton,
+    unsafe extern "C" fn change_trampoline<T, F: Fn(&Self, ScrollType) + 'static>(this: *mut GtkSpinButton,
                                            scroll: ScrollType,
-                                           f: &&(Fn(&SpinButton, ScrollType) + 'static)) {
-        f(&from_glib_borrow(this), scroll)
+                                           f: &F)
+    where T: IsA<SpinButton> {
+        f(&from_glib_borrow(this).unsafe_cast(), scroll)
     }
 
     unsafe extern "C" fn input_trampoline(this: *mut GtkSpinButton,
                                           new_value: *mut c_double,
-                                          f: &&(Fn(&SpinButton) -> Option<Result<f64, ()>> + 'static))
-                                          -> c_int {
-        match f(&from_glib_borrow(this)) {
+                                          f: &F)
+                                          -> c_int
+    T: IsA<SpinButton> {
```

Comment:
No `where` ?

Reply:
Yeah I can't compile things currently until @GuillaumeGomez is done with his regen PRs :)

Reply:
*3 and many other errors,

Reply:
Yeah we can fix and merge this one once the other things are done :)